server.go

```diff
@@ -243,6 +245,60 @@ func newServer(listenAddrs []string, chanDB *channeldb.DB, cc *chainControl,
 	}
 	s.htlcSwitch = htlcSwitch
 
+	// If enabled, use either UPnP or NAT-PMP to automatically configure
+	// port forwarding for users behind a NAT.
+	switch {
+	case cfg.Upnp:
+		srvrLog.Info("Attempting to automatically configure port " +
+			"forwarding rules using UPnP")
+
+		upnp, err := discoverUPnP()
+		if err != nil {
+			srvrLog.Errorf("Unable to discover a UPnP enabled "+
+				"device on the local network: %v", err)
+			break
+		}
+
+		s.natTraversal = upnp
+	case cfg.NatPmp:
+		srvrLog.Info("Attempting to automatically configure port " +
+			"forwarding rules using NAT-PMP")
+
+		pmp, err := discoverNATPMP()
+		if err != nil {
+			srvrLog.Errorf("Unable to discover a NAT-PMP enabled "+
+				"device on the local network: %v", err)
+			break
+		}
+
+		s.natTraversal = pmp
+	}
+
+	if s.natTraversal != nil {
+		ip, err := s.natTraversal.ExternalIP()
```

Comment:
Since the bulk of this body is done against generic `NATTraversal` instances, this might be useful as a package helper function, like:
```golang
ip, err := nat.ConfigurePortForward(port, s.natTraversal)
if err != nil {
    // log err to error
} else {
    // log success to info
    s.lastIP = ip
    cfg.ExternalIPs = append(cfg.ExternalIPs, ip.String())
}
```

Reply:
Fixed.